File: Fyp/bidbazar/lib/data/models/user_model.dart
class userModel {
  String? sId;
  String? fullname;
  String? email;
  String? mobile;
  String? cnic;
  String? address;
  String? password;
  String? usertype;
  String? id;
  bool? block;
  bool? verification;
  List<String>? cnicimages;
  List<String>? profileimages;

  userModel(
      {this.sId,
      this.fullname,
      this.email,
      this.mobile,
      this.cnic,
      this.address,
      this.password,
      this.usertype,
      this.id,
      this.block,
      this.verification,
      this.profileimages,
      this.cnicimages
      });

  userModel.fromJson(Map<String, dynamic> json) {
    //return method
    sId = json['_id'];
    fullname = json['fullname'];
    email = json['email'];
    mobile = json['mobile'];
    cnic = json['cnic'];
    address = json['address'];
    password = json['password'];
    usertype = json['usertype'];
    id = json['id'];
    block = json['block'];
    verification = json['verification'];
    profileimages = json['profileimages'].cast<String>();
    cnicimages = json['cnicimages'].cast<String>();
  }

  Map<String, dynamic> toJson() {
    // set method
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['_id'] = this.sId;
    data['fullname'] = this.fullname;
    data['email'] = this.email;
    data['mobile'] = this.mobile;
    data['cnic'] = this.cnic;
    data['address'] = this.address;
    data['password'] = this.password;
    data['usertype'] = this.usertype;
    data['id'] = this.id;
    data['block'] = this.block;
    data['verification'] = this.verification;
    return data;
  }
}

File: backend/src/models/user_model.js
const mongoose = require('mongoose');
const uuid = require('uuid');
const bcryptjs = require("bcryptjs");


///for id generate auto we have install package 
// Declare the Schema of the Mongo model

var userSchema = new mongoose.Schema({
    id:{
        type:String,
        unique:true,
        index:true,
    },   
    fullname:{
        type:String,
        required:true,
        index:true,
    },
    email:{
        type:String,
        required:true,
        unique:true,
    },
    mobile:{
        type:String,
        required:true,
        unique:true,
    },
    cnic:{
        type:String,
        required:true,
        unique:true,
    },
    address:{
        type:String,
        default:"",
    },
    password:{
        type:String,
        required:true,

    },
    usertype:{
        type:String,
        required:true,
    },   
    block:{
        type:Boolean,
        required:true,
        default:false,
    },
    verification:{
        type:Boolean,
        required:true,
        default:false
    }, 
    cnicimages:{
        type:Array,
        default:[]
    },
    profileimages:{
        type:Array,
        default:[]
    },
},);

userSchema.pre("save",function(next){//save ke time kiya hoga
        this.id=uuid.v1();        
        this.password=bcryptjs.hashSync(this.password,8);
        next();
});


userSchema.pre(['update','findOneAndUpdate','updateOne'],function(next){
const update= this.getUpdate();
    delete update._id;
    delete update.id;
    // delete this.password;
    next();
});

const UserModel =mongoose.model('User', userSchema);//users collection

//Export the model
module.exports = UserModel;